src/components/SearchForm.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import { Panel, Button } from "library-simplified-reusable-components";
+import { SearchIcon } from "@nypl/dgx-svg-icons";
+import Form from "./reusables/Form";
+import Input from "./reusables/Input";
+
+export interface SearchFormOwnProps {
+  search: (data: FormData) => void;
+  text: string;
+  inputName: string;
+  clear?: () => any;
+  term?: string;
+  resultsCount?: number;
+}
+
+export interface SearchFormState {
+  searchTerm: string;
+}
+
+export default class SearchForm extends React.Component<SearchFormOwnProps, SearchFormState> {
+  constructor(props: SearchFormOwnProps) {
+    super(props);
+    this.updateSearchTerm = this.updateSearchTerm.bind(this);
+    this.message = this.message.bind(this);
+    this.state = { searchTerm: "" };
+  }
+
+  render(): JSX.Element {
+    let input = (
+      <Input
+        name={this.props.inputName}
+        callback={this.updateSearchTerm}
+      />
+    );
+
+    let form = (
+      <Form
+        onSubmit={this.props.search}
+        content={input}
+        buttonContent={<span>Search <SearchIcon /></span>}
+        className="inline"
+        disableButton={!this.state.searchTerm.length}
+        successText={this.message()["success"]}
```

Comment:
Make `this.message()` into a variable, that way you don't call the function twice for each separate value.